functions/validation/validation.js

```diff
@@ -0,0 +1,172 @@
+const {Timestamp} = require("firebase-admin/firestore");
+const ERRORS = require("../utils/errors.js");
+const REGEX = require("../utils/regex.js");
+
+/**
+ * Helper function to check the release name format.
+ * The format of release names must be "M<releaseNumber>"
+ *
+ * @param {string} name - The release name.
+ * @return {bool} - True if the release name is valid.
+ */
+function isValidReleaseName(name) {
+  return REGEX.RELEASE_NAME.test(name);
+}
+
+/** Helper function to check that a string represents a valid date.
+ *
+ * @param {string} dateString - A date in string format.
+ * @return {bool} - True if the string is a valid date.
+ */
+function isValidDate(dateString) {
+  if (typeof dateString !== "string") {
+    return false;
+  }
+  const date = new Date(dateString);
+  return !isNaN(date.getTime());
+}
+
+/**
+ * Validates that a release object is in a valid form.
+ *
+ * @param {Object} release - The release to validate.
+ * @return {Array} errors - A list of errors from the validation of the release.
+ */
+function validateRelease(release) {
```

Comment:
A more functional approach would be like such:
```js
function validateRelease(release) {

  const nameErrors = validateReleaseName(release);
  const operatorErrors = validateReleaseOperator(release);
  const dateErrors = validateReleaseDates(release);

  return [...nameErrors, ...operatorErrors, ...dateErrors];
}

Reply:
That's definitely much better!

Resolved in https://github.com/firebase/firebase-release-dashboard/pull/8/commits/a32e049b16404c7818f488231642a11b343b3068